pkg/discovery/service.go

```diff
@@ -143,6 +143,8 @@ func (discovery *Discovery) refresh(ctx context.Context) error {
 			return ctx.Err()
 		}
 
+		// TODO: add method to update ping.LastIp? Or is it done above in cache.Put?
+
```

Comment:
This should be done automatically with the FetchInfo function, which is then used in UpdateNodeInfo.